### PR TITLE
API integration tests boiler plate

### PR DESCRIPTION
**Pull request by individual-it**
alternative approach to #32 see discussion there

an approach that does not need to run the application itself is in #54 
**only merge this PR OR #54**

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -118,6 +118,7 @@ def testing(ctx):
         'image': 'webhippie/golang:1.13',
         'pull': 'always',
         'commands': [
+          './bin/ocis-hello server &',
```
Comment (tboerger) [inline .drone.star]: This is not needed for API testing, beside that if we would do that this should be a separate detached step.

File: tests/integration/hello_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package proto
+
+import (
+	"github.com/stretchr/testify/assert"
```
Comment (tboerger) [inline tests/integration/hello_test.go]: Libraries from standard lib firtst, than an empty line, than github/whatever imports.

**Comment by individual-it** [comment]
@tboerger I could not find a way to test through the handles see my comment in #32 https://github.com/owncloud/ocis-hello/pull/32#issuecomment-568678950
This seems not to be supported by swagger directly and the only functional approach looks hacky to me see: https://goswagger.io/faq/faq_testing.html
Could you guide me, how else we can achieve it?

File: tests/integration/hello_test.go (hunk 2 above)
Comment (individual-it) [reply tests/integration/hello_test.go]: done in 83cea2847da28967f0fba35ebd6d03eb3635d805

**Comment by tboerger** [comment]
What do you want to do with swagger? I don't get it...

**Comment by individual-it** [comment]
I thought swagger.io is used to generate the API code of the project

**Comment by tboerger** [comment]
> I thought swagger.io is used to generate the API code of the project

There is a Swagger definition generated for the RESTful API and we are generating a simple javascript library from it, that's all. The swagger definition is not really needed to write API tests. BEside that, not only the HTTP API should be tested, the GRPC API should also be tested.

**Comment by individual-it** [comment]
I think I worked it out finally, please have a look at https://github.com/owncloud/ocis-hello/pull/54

**Comment by individual-it** [comment]
closing in favour of #54

**Pull request closed without merging**